test/datacollector.UnitTests/DataCollectionAttachmentManagerTests.cs

```diff
@@ -73,6 +73,17 @@ public void InitializeShouldSetCorrectGuidAndOutputPath()
             Assert.AreEqual(Path.Combine(TempDirectoryPath, this.sessionId.Id.ToString()), this.attachmentManager.SessionOutputDirectory);
         }
 
+        [TestMethod]
+        public void InitializeShouldNotCreateDirectoryIfNotHaveAccessToFolder()
```

Comment:
This test seems like it tests an obvious thing, and that we will have no idea what is being tested, when we re-visit it in a month or so. Not sure how to test it better though. So I would at least add a comment to avoid looking for the PR via blame when this breaks.

Reply:
I will change it to checking if user has access, so it will not throw exception, if he doesn't it will throw. That logic should be give more clear understanding.

Reply:
Thought that checking permission and then running test logic will not give any benefit, as it will be almost the same logic and code repetition. For now I think we can not having test for this part of code, as it also fails in our pipeline.